polyfactory/collection_extender.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+import random
+from abc import ABC
+from typing import Any, TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import GenericAlias  # type: ignore
+
+
+class CollectionExtender(ABC):
+    __types__: tuple[type, ...]
+
+    @staticmethod
+    def _extend_type_args(type_args: tuple[Any, ...], number_of_args: int) -> tuple[Any, ...]:
+        raise NotImplementedError
+
+    @classmethod
+    def _subclass_for_type(cls, annotation_alias: GenericAlias) -> type[CollectionExtender]:
+        for subclass in cls.__subclasses__():
+            try:
+                is_subclass = issubclass(annotation_alias.__origin__, subclass.__types__)
+            except TypeError:
+                # e.g., Union, or other non-class types
+                return FallbackExtender
+            if is_subclass:
```

Comment:
Move this and the next two lines into the try block. Also, use the walrus operator in the if statement